[Add]: Relationship between PriorityClass and other k8s components

pandeyshubham03 opened the pull request:
**Notes for Reviewers**

This PR fixes #
1. Adds relationship between PriorityClass and other k8s components.

https://github.com/meshery/meshery/assets/128715606/84ae25eb-b32b-4b7c-9cdc-d1afe3a29639


**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1090,6 +1090,40 @@
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "PriorityClass in Kubernetes defines scheduling priority for Pods."
```
Please describe the relationship in addition to the component.

RipulHandoo commented:
@pandeyshubham03 PriorityClass can indeed have relationships with various Kubernetes components, including Deployments, Pod Template and others.

pandeyshubham03 commented:
I have added the description for the relationships added @leecalcote and I have also added all the components that could have a relationships with PriorityClass @RipulHandoo.

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1327,6 +1327,118 @@
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "PriorityClass in Kubernetes assigns importance levels to Pods, influencing scheduling decisions. The name of the PriorityClass is referenced by other k8s components"
```
Let's not abbreviate "Kubernetes" or if we do, make sure that we abide by their branding request that "K8s" be capitalized. Having said that, let's not abbreviate Kubernetes. The PriorityClass name is referenced by other components... for what purpose?

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1327,6 +1327,118 @@
+              "mutatedRef": [
+                ["settings", "spec", "spec", "priorityClassName"]
+              ],
+              "description": "PodTemplate specifies the name of priorityClass, guiding Kubernetes controller to assign priority to Pods created from it, influencing scheduling behavior accordingly."
```
"Kubernetes controller" - make this plural.

"created from it" - "it" who?

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1327,6 +1327,118 @@
+              "mutatedRef": [
+                ["settings", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "StatefulSets manage stateful applications. PriorityClass can be applied to StatefulSets' Pods to prioritize them during scheduling."
```
"them" - to what component are you referring?

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1327,6 +1327,118 @@
+              "mutatedRef": [
+                ["settings", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "DaemonSets ensure all nodes run a copy of a Pod. PriorityClass assigns importance, impacting scheduling decisions for DaemonSet-managed Pods."
```
"impacting" - in what way?

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1327,6 +1327,118 @@
+              "mutatedRef": [
+                ["settings", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "PriorityClass in Kubernetes influences Job scheduling, ensuring high-priority Jobs execute first, critical for resource allocation and workload management."
```
Rephrase - ", critical for resource allocation and workload management."

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1327,6 +1327,118 @@
+              "mutatedRef": [
+                ["settings", "spec", "jobTemplate", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "CronJobs can use PriorityClass to prioritize scheduled job execution, ensuring critical tasks are handled with precedence in Kubernetes."
```
CronJobs can or do use?

"handled with precedence in Kubernetes." or "ensuring CronJobs are scheduled for execution in accordance with the precedence specified in the associated PriorityClass."?

leecalcote commented on server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:
```diff
@@ -1327,6 +1327,118 @@
+              "mutatedRef": [
+                ["settings", "spec", "jobTemplate", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "CronJobs can use PriorityClass to prioritize the scheduled job execution, ensuring CronJobs are scheduled for execution in accordance with the precedence specified in the associated PriorityClass."
```
Nice!
